src/ucp/core/ucp_context.c

```diff
@@ -1577,14 +1601,15 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
         }
     }
 
-    /* Update registration memory domain map for host memory type taking into
-     * account result of uct_md_mem_reg. */
-    status = ucp_mem_reg_md_map_update(context);
+    status = ucp_mm_init_alloc_md_index(context);
```

Comment:
can we do this on-demand and not during initialization?

Reply:
done